### DISCUSSION TITLE
How to lock the cursor position on the screen?

### DISCUSSION THREAD

**Discussion by bogdan-ov** [HaxeFlixel/flixel#2786] [Q&A] [answered]
I want to lock the cursor position on the game screen (in the center or just don't let the cursor go outside the window)
How can I do this?
**Also** I want to listen the mouse movement if possible

**Comment by bwkam** [comment]
I recommend checking out https://api.haxeflixel.com/flixel/input/mouse/FlxMouse.html

- You can also have access to the main pointer by [`FlxG.mouse`](https://api.haxeflixel.com/flixel/FlxG.html#mouse) which is an instance of [`FlxMouse`](https://api.haxeflixel.com/flixel/input/mouse/FlxMouse.html)

**Comment by bogdan-ov** [comment, marked as answer]
Well... I found the solution by myself

```haxe
// Put it somewhere in `update` function
public function update(elapsed:Float) {
    FlxG.stage.application.window.warpMouse(0, 0);
}
```

**Comment by TheScratchTeen** [comment]
I think with:
```Haxe
var mouseChange = FlxPoint.get(0, 0);
function create() {
    FlxG.mouse.visible = false;
    FlxG.stage.application.window.warpMouse(0, 0);
}
function update() {
    mouseChange.set(FlxG.mouse.x, FlxG.mouse.y);
    FlxG.stage.application.window.warpMouse(0, 0);
}
```

**Reply by Geokureli** [reply]
The author already answered their own question. I would avoid trying to engage with 3+ year old discussions, The author is likely long gone, and I'll probably just lock them
